conans/model/options.py

```diff
@@ -139,6 +139,8 @@ def rm_safe(self, field):
             delattr(self, field)
         except ConanException:
             pass
+        except KeyError:
```

Comment:
Instead of adding this, the best would try to start with a test that breaks, if this is not here. Then we can take it from there, and see the best solution, but better first to illustrate the problem.

Reply:
I have pushed the minimal changes that make it fail and removed this